pkg/controller/master/backup/restore.go

```diff
@@ -148,6 +166,35 @@ func (h *RestoreHandler) RestoreOnChanged(key string, restore *harvesterv1.Virtu
 	return nil, h.updateStatus(restore, backup, vm, isVolumesReady)
 }
 
+// RestoreOnRemove delete VolumeSnapshotContent which is created by restore controller
+// Since we would like to prevent LH Backups from being removed when users delete the VM,
+// we use Retain policy in VolumeSnapshotContent.
+// We need to delete VolumeSnapshotContent by restore controller, or there will have remaining VolumeSnapshotContent in the system.
+func (h *RestoreHandler) RestoreOnRemove(key string, restore *harvesterv1.VirtualMachineRestore) (*harvesterv1.VirtualMachineRestore, error) {
+	if restore.Status == nil {
```

Comment:
Should it check the existence of restore as well? Most of `OnRemove` callbacks in Harvester did that.
```suggestion
	if restore == nil || restore.Status == nil {
```

Reply:
Updated it. Thank you.